go-controller/pkg/util/multi_network.go

```diff
@@ -295,19 +295,23 @@ func ParseNetConf(netattachdef *nettypes.NetworkAttachmentDefinition) (*ovncnity
 //
 // Return value:
 //    bool: if this Pod is on this Network; true or false
-//    *nettypes.NetworkSelectionElement: all NetworkSelectionElement that pod is requested for the specified network
+//    map[string]*nettypes.NetworkSelectionElement: all NetworkSelectionElement that pod is requested
+//        for the specified network, key is NADName. Note multiple NADs of the same network are allowed
+//        on one pod, as long as they are of different NADName.
 //    error:  error in case of failure
-func PodWantsMultiNetwork(pod *kapi.Pod, nInfo NetInfo) (bool, *nettypes.NetworkSelectionElement, error) {
-	var validNADName string
-
+func PodWantsMultiNetwork(pod *kapi.Pod, nInfo NetInfo) (bool, map[string]*nettypes.NetworkSelectionElement, error) {
+	networkSelections := map[string]*nettypes.NetworkSelectionElement{}
 	podDesc := fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
 	if !nInfo.IsSecondary() {
 		network, err := GetK8sPodDefaultNetwork(pod)
 		if err != nil {
 			// multus won't add this Pod if this fails, should never happen
 			return false, nil, fmt.Errorf("error getting default-network's network-attachment for pod %s: %v", podDesc, err)
 		}
-		return true, network, nil
+		if network != nil {
```

Comment:
wouldn't a pod always want the primary network, or not have any choice whether or not they want the primary network?

Reply:
I thought we discussed that, and we conclude that we assume the default(primary network) is OVN, regardless if its "v1.multus-cni.io/default-network" annotation actually request a non-OVN network (that is not supported).